packages/mermaid/src/diagrams/xychart/parser/xychart.jison

```diff
@@ -116,6 +117,8 @@ plotData
 commaSeparatedNumbers
   : NUMBER_WITH_DECIMAL COMMA commaSeparatedNumbers                { $$ = [Number($NUMBER_WITH_DECIMAL), ...$commaSeparatedNumbers] }
   | NUMBER_WITH_DECIMAL                                           { $$ = [Number($NUMBER_WITH_DECIMAL)] }
+  | NAN COMMA commaSeparatedNumbers                               { $$ = [Number($NAN), ...$commaSeparatedNumbers] }
+  | NAN                                                           { $$ = [Number($NAN)] }
```

Comment:
I feel like this is a wrong place to represent NaN, because `commaSeparatedNumbers` represents the list, and does not necessarily have to deal with all the possible values. Adding more things apart from numbers will inflate the grammar. My suggestion is to combine `NUMBER_WITH_DECIMAL` and `NAN` into a separate rule called `optionalValue`, but my suggestion is optional, I guess :smiley: